### ISSUE TITLE
Add advisories from SRC:CLR

### ISSUE THREAD

**Issue by reedloden** [rubysec/ruby-advisory-db#238]
Just a todo list I figured I should put somewhere more public... Need to add advisories for all these:

ruby_rncryptor / ruby_rncryptor_secured -- https://srcclr.com/security/timing-attacks/ruby/s-1938

spina -- https://srcclr.com/security/cross-site-request-forgery-csrf/ruby/s-1686

logstash-core -- https://srcclr.com/security/factoring-attack-rsa-export-keys-freak/ruby/s-1745
https://srcclr.com/security/man-middle-mitm-attacks/ruby/s-1798

facter -- https://srcclr.com/security/disclosure-amazon-ec2-iam-instance/ruby/s-1508
https://srcclr.com/security/elevation-privileges-untrusted-search/ruby/s-1586

kafo -- https://srcclr.com/security/world-readable-permissions-as-default/ruby/s-740

puppet -- https://srcclr.com/catalog/search#query=type:vulnerability%20puppet

**Comment by VanessaHenderson** [comment]
refinerycms-core 
https://srcclr.com/security/cross-site-scripting-xss-through-title/ruby/s-2168
https://srcclr.com/security/cross-site-scripting-xss-through-alt/ruby/s-2169
https://srcclr.com/security/stored-cross-site-scripting-xss-title/ruby/s-2170
https://srcclr.com/security/cross-site-request-forgery-csrf/ruby/s-2171

**Comment by VanessaHenderson** [comment]
paperclip -- https://srcclr.com/security/denial-service-dos-through-excessive/ruby/s-2242

**Comment by VanessaHenderson** [comment]
admin-upmin, upmin and shoppe -- https://srcclr.com/security/cross-site-request-forgery-csrf-due-to/ruby/s-2266
devise_invitable -- https://srcclr.com/security/cross-site-request-forgery-csrf/ruby/s-2272

**Comment by reedloden** [comment]
activeadmin -- https://srcclr.com/security/cross-site-scripting-xss-through-modal/ruby/s-2276

**Comment by skorth** [comment]
@reedloden regarding spina ruby gem https://github.com/rubysec/ruby-advisory-db/pull/250

**Comment by jasnow** [comment]
Not sure https://srcclr.com is an active web site.

**Comment by VanessaHenderson** [comment]
You are correct, SourceClear was purchased by Veracode. The easiest way to find the links now would be to search the s-* number at https://sca.analysiscenter.veracode.com/vulnerability-database
A lot of them will be privated unless you have a Veraode SCA account, otherwise it is just basic information available.

IE; Activeadmin that @reedloden posted is here: https://sca.analysiscenter.veracode.com/vulnerability-database/security/cross-site-scripting-xss-through-modal-dialog/ruby/sid-2276/summary

**Comment by jasnow** [comment]
@VanessaHenderson  - Thanks for the education.

**Comment by jasnow** [comment]
> h#query=type:vulnerability%20puppet

Need a list of specific "puppet" URLs.

**Comment by jasnow** [comment]
> admin-upmin, upmin and shoppe -- https://srcclr.com/security/cross-site-request-forgery-csrf-due-to/ruby/s-2266 devise_invitable -- https://srcclr.com/security/cross-site-request-forgery-csrf/ruby/s-2272

It appears that "s-2272" for devise_invitable may have be withdrawn if this is the same advisory: https://github.com/advisories/GHSA-wj5j-xpcj-45gc

**Comment by VanessaHenderson** [comment]
It looks like SourceClear itself hasn't withdrawn the item, I wonder why GitHub withdrew it...

**Comment by VanessaHenderson** [comment]
Ah yeah that'd probably do it. SourceClear/Veracode adds vulnerabilities that aren't necessarily assigned CVEs, basing on code fixes etc instead of solely CVE

**Comment by jasnow** [comment]
## After research the 16 items in this issue again today, here are my updates:

 -1- Two (ruby_rncryptor, ruby_rncryptor_secured) dead gems/not released since 2017
 -2- spina (Found 2 more refs to gems/spina/CVE-2015-4619.yml/closed) -3- See gems/logstash-core/CVE-2015-5378.yml/closed
 -4- See gems/logstash-core/CVE-2015-5619.yml/closed
 -5- See gems/facter/CVE-2015-1426.yml/closed
 -6- (under facter) See gems/puppet/CVE-2014-3248.yml/closed
 -7- See gems/kafo/CVE-2014-0135.yml/closed
 -8- puppet (TOO BROAD/dropped)
 -9-to-12- refinerycms-core (No additional data and no CVEsor GHSAs/closed)
 -13- paperclip (Found fix, but no advisory/will add to #238/closed)
 -14- Three (admin-upmin, upmin, shoppe) dead gems - not released since 2015/closed
 -15- devise_invitable (GHSA-wj5j-xpcj-45gc withdrawn/Closed)
 -16- activeadmin (Found fix, but no advisory/will add to #238/closed)

## References for 2 gems

 -13- Paperclip (Mar 15, 2016 comment) References:
       - Release 4.2.2 to 4.3.5 had DOS.
       - https://sca.analysiscenter.veracode.com/vulnerability-database/security/denial-service-dos-through-excessive/ruby/sid-2242/summary  
       - https://thoughtbot.com/blog/prevent-spoofing-with-paperclip
       - https://seclists.org/oss-sec/2016/q1/640
       - https://web.archive.org/web/20250821081556/https://github.com/thoughtbot/paperclip/pull/2126)
         (Backport fix for excessive logging from media type spoof detector)

 -16- Activeadmin (Apr 3, 2016 comment) References:
       - https://sca.analysiscenter.veracode.com/vulnerability-database/security/cross-site-scripting-xss-through-modal-dialog/ruby/sid-2276/summary
      - https://github.com/activeadmin/activeadmin/issues/5198
      - https://web.archive.org/web/20200000000000*/https://github.com/activeadmin/activeadmin/pull/3811

## Expect to close this issue in a couple of days.

**Comment by jasnow** [comment]
Acting on my previous comment, I am closing this issue.
